src/gobapi/graphql_streaming/graphql2sql/graphql2sql.py

```diff
@@ -426,34 +455,61 @@ def _join_relation_table(self, src_relation: dict, relation_name: str, rel_table
         :return:
         """
         rel_left = 'src' if not is_inverse else 'dst'
-        relation_table = f"{self.SCHEMA}.mv_{relation_name}"
+        rel_right = 'dst' if not is_inverse else 'src'
+        relation_table = self._relation_table_name(relation_name)
+
+        def join_reltable(reltable_alias: str, reltable_side=rel_left, join_with_table='src'):
+            join_relation = src_relation if join_with_table == 'src' else dst_relation
 
-        def join_filters(table_alias: str):
             filters = [
-                f"{table_alias}.{rel_left}_id = {src_relation['alias']}.{FIELD.ID}"
+                f"{reltable_alias}.{reltable_side}_id = {join_relation['alias']}.{FIELD.ID}"
             ]
 
             if not is_inverse and src_value_requested:
-                match_src_value_with = self._add_srcvalue_selection(src_relation, src_attr_name, is_many)
-                filters.append(f"{table_alias}.{FIELD.SOURCE_VALUE} = {match_src_value_with}")
+                match_src_value_with = self._add_srcvalue_selection(join_relation, src_attr_name, is_many)
+                filters.append(f"{reltable_alias}.{FIELD.SOURCE_VALUE} = {match_src_value_with}")
 
-            if src_relation['has_states']:
-                filters.append(f"{table_alias}.{rel_left}_volgnummer = {src_relation['alias']}.{FIELD.SEQNR}")
+            if join_relation['has_states']:
+                filters.append(f"{reltable_alias}.{reltable_side}_volgnummer = {join_relation['alias']}.{FIELD.SEQNR}")
 
             return " AND ".join(filters)
 
         if arguments.get('first'):
+            if arguments.get('sort'):
+                sort = [s.replace('_desc', ' DESC')
+                        if s.endswith('_desc')
+                        else s.replace('_asc', ' ASC')
+                        for s in arguments['sort']]
+            else:
+                sort = [FIELD.GOBID]
+            order_by = ', '.join([f"{dst_relation['alias']}.{s}" for s in sort])
+
+            first = int(arguments['first'])
+            match_row_number = f"{rel_table_alias}.row_number <= {first}" if first > 1 \
```

Comment:
- Row_number is used a few lines below, to limit the number of relations returned. E.g. in meetbouten export to get the first meting (order by datum asc and limit 1) and last meting (order by datum desc and limit 1) back in one GraphQL query. This wasn't implemented properly before.
- Could extract first into a new method, but that would involve a few more parameters than ideal and then the context is gone. Would need to extract the join_reltable function as well, which would lose context, would need to pass the rel_left, rel_right parameters, etc. Would not make it any clearer imo. Also; the 'first' branch is most of the method (the else-part is just one line), so it would just be moving that stuff into some other place. The cc of this method is still fairly low - I don't think reducing this would make the method easier to understand/read. Now it just fits neatly onto one screen. It's a bit lengthy, but I think it works.